mep/books/oclc.py

```diff
@@ -92,26 +92,102 @@ def marc_records(self) -> List[pymarc.record.Record]:
         return list(filter(None, pymarc.parse_xml_to_array(bytestream)))
 
 
-def oclc_uri(marc_record: pymarc.record.Record) -> str:
-    """Generate the worldcat URI for an OCLC MARC record"""
-    return 'http://www.worldcat.org/oclc/%s' % marc_record['001'].value()
-
 
 #: schema.org RDF namespace
 SCHEMA_ORG = rdflib.Namespace('http://schema.org/')
 
 
+class RdfResource(rdflib.resource.Resource):
+    # Extend rdflib resource to fix string method. This is based
+    # on their code, but actually works in python 3
+
+    def __str__(self):
+        return str(self._identifier)
+
+class WorldCatEntity:
+    '''Entity for a single WorldCat record, with support for
+    loading linked data and accessing values.'''
+
+    rdf_resource = None
+
+    @staticmethod
+    def oclc_uri(marc_record: pymarc.record.Record) -> str:
+        """Generate the worldcat URI for an OCLC MARC record"""
+        return 'http://www.worldcat.org/oclc/%s' % marc_record['001'].value()
+
+    def __init__(self, marc_record: pymarc.record.Record, session: requests.Session):
+        self.marc_record = marc_record
+        self.session = session
+        self.item_uri = WorldCatEntity.oclc_uri(self.marc_record)
+        graph = self.get_oclc_rdf()
+        if graph:
+            # initialize an rdflib.resource
+            self.rdf_resource = RdfResource(graph, rdflib.URIRef(self.item_uri))
+
+    def get_oclc_rdf(self) -> rdflib.Graph:
+        '''Load the RDF for the OCLC URI for the MARC record for this
+        entity.'''
+
+        graph = rdflib.Graph()
+        # control field in 001 is OCLC identifier, used for URIs
+        response = self.session.get("%s.rdf" % self.item_uri)
+        if response.status_code == requests.codes.ok:
+            graph.parse(data=response.content.decode())
+            return graph
+
+        # log an error for any other status
+        logger.error('Error loading OCLC record as RDF %s => %d',
+                     self.item_uri, response.status_code)
+
+    @property
+    def work_uri(self) -> str:
+        '''OCLC Work URI for this item'''
+        value = self.rdf_resource.value(SCHEMA_ORG.exampleOfWork)
+        return str(value) if value else None
+
+    @property
+    def item_type(self) -> str:
```

Comment:
Is there a possibility of this ever not finding an `item_type`? I don't know the schema that well, so that's my only concern here. 

In re: `item_type` as a many-to-many, I think it depends on what we want: a 'main' type for the work, that would be easily filterable and displayable, or whether we want to capture *everything*. My inclination is simpler is better unless we have a compelling reason.

Reply:
OCLC records all have an RDF:type; I think RDF records generally do, but since we're working with OCLC records specifically I think it's safe.

It can be multiple, but I think we just want one main non-generic type (which is why I'm filtering out the more generic creative work type).

Reply:
@bwhicks oh, I see now what you mean about many-to-many: I misspoke (-wrote), I meant perhaps it should be a model and `Item` would have a `ForeignKey`. I think that may have some advantages for admin users working with the data, since we want this field to be something they can edit and manage (or at least we did previously), and I don't think we want them pasting in URIs. A model would let us track the URIs but display human-readable labels. (Although perhaps we don't care about storing the URIs.)

Seems like something to look at with the revised results (after filtering by item type) and maybe discuss with the team, since we have a separate issue for that feature (#224).

Reply:
I agree, and that makes eminent sense (both in approach and in waiting for the separate feature to talk to the team).